fms/utils/generation.py

```diff
@@ -1,7 +1,104 @@
-from typing import Any, Callable, List, MutableMapping, Optional, Union
+import inspect
+from typing import Any, Callable, Dict, List, MutableMapping, Optional, Tuple, Union
 
 import torch
+import torch.nn as nn
 import torch.nn.functional as F
+from torch._dynamo import OptimizedModule
+
+
+def __prepare_list_input(
+    input_ids_list: List[torch.Tensor], model: Union[Callable, torch.nn.Module]
+) -> Tuple[torch.Tensor, MutableMapping[str, Any]]:
+    """
+    Convert the list of Tensors to a rectangular tensor. Return extra kwargs for the position_ids and mask, since this
+    will be required to properly handle the rectangular tensor for certain models.
+    """
+    min_len = min([seq.size(0) for seq in input_ids_list])
+    max_len = max([seq.size(0) for seq in input_ids_list])
+
+    if isinstance(model, OptimizedModule):
+        forward_function = model._orig_mod.forward
```

Comment:
this works as advertised? I know I told you about it, but first time I see it used

Reply:
This is working well, I tested it

Reply:
If we use nn.module.compile I think we could rely on `_call_impl` here, which might be safer.

Reply:
it might also be preferable to move this to a `_get_model_kwargs(model)`

Reply:
Switched to `model.compile` as it worked well and is a safer option.